packages/react-native-worklets/src/shareables.ts

```diff
@@ -406,7 +436,7 @@ function clonePlainJSObject<T extends object>(
 function cloneRegExp<T extends RegExp>(value: T): ShareableRef<T> {
   const pattern = value.source;
   const flags = value.flags;
-  const handle = makeShareableCloneRecursive({
+  const handle = cloneInitializer({
```

Comment:
Since now we have `cloneInitializer` function we can change it so that it accepts only the factory worklet instead of an object with `__init` property and also completely remove `__init` property from our logic in `ShareableInitializer` in C++.